### PR TITLE
Feature/instagram signup

### PR DESCRIPTION
**Pull request by Manan-Sharma-5**
## Summary by CodeRabbit

- **New Features**
	- Introduced a component for selecting a primary social media account.
	- Added a screen for users without connected social accounts, prompting them to connect.
	- Enhanced authentication flow to include Instagram login alongside Facebook.
	- Integrated Instagram login functionality within the connected socials component.
	- Implemented a context provider for managing user social media profiles.
	- Added a button for connecting Facebook accounts with updated labels.

- **Bug Fixes**
	- Improved error handling and user feedback during social media updates.

- **Documentation**
	- Updated comments and structure for clarity in various components.

- **Refactor**
	- Streamlined logic for displaying user profile images and handling social media sign-in processes.

### CODE REVIEW

File: app/(main)/(screens)/questions.tsx [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
 import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
-import { faArrowRight, faChevronLeft } from "@fortawesome/free-solid-svg-icons";
+import {
+  faArrowRight,
+  faCheck,
+  faChevronLeft,
+} from "@fortawesome/free-solid-svg-icons";
 import Select from "@/components/ui/select";
 import { Text, View } from "@/components/theme/Themed";
+import { useSocialContext } from "@/contexts/social-context.provider";
```
Comment (dev-rathore) [inline app/(main)/(screens)/questions.tsx:18]: We can shift this to the contexts folder index.ts file and import it from there to maintain consistency.

File: contexts/auth-context.provider.tsx [hunk 2]
```diff
@@ -95,22 +107,22 @@ export const AuthContextProvider: React.FC<PropsWithChildren> = ({
...
-      await analyticsLogEvent('signed_in', {
+      await analyticsLogEvent("signed_in", {
         id: userCredential.user.uid,
         name: userCredential.user.displayName,
         email: userCredential.user.email,
       });
 
       // For existing users, redirect to the main screen.
-      router.replace("/collaborations");
+      // router.replace("/collaborations");
```
Comment (dev-rathore) [inline contexts/auth-context.provider.tsx:125]: Why did we comment on this?

**Review by dev-rathore — CHANGES_REQUESTED** [review]

File: contexts/auth-context.provider.tsx (hunk 2 above)
Comment (Manan-Sharma-5) [reply contexts/auth-context.provider.tsx:125]: Because now we want to first check if user has socials then only we will move then to collaborations

Comment (dev-rathore) [reply contexts/auth-context.provider.tsx:125]: So can we remove it instead of commenting?

Comment (Manan-Sharma-5) [reply contexts/auth-context.provider.tsx:125]: Ok

**Review by dev-rathore — APPROVED** [review]

**Pull request merged**